src/main/scala/firrtl/transforms/InlineBooleanExpressions.scala

```diff
@@ -84,10 +84,18 @@ class InlineBooleanExpressions extends Transform with DependencyAPIMigration {
     /** Whether or not an can be inlined
       * @param refExpr the expression to check for inlining
       */
-    def canInline(refExpr: Expression): Boolean = {
-      refExpr match {
-        case _: Mux => false
-        case _ => refExpr.tpe == Utils.BoolType
+    def canInline(refExpr: Expression, outerExpr: Option[Expression]): Boolean = {
+      val contextInsensitiveDetOps: Set[PrimOp] = Set(Lt, Leq, Gt, Geq, Eq, Neq, Andr, Orr, Xorr)
```

Comment:
It's not quite the same as self-determined expressions -- both self-determined ops and "operands are self-determined" factors in. I originally called it `selfDeterminedOps` but changed that since IMO that was confusing.